ipwb/util.py

```diff
@@ -63,20 +56,18 @@ def isDaemonAlive(daemonMultiaddr=IPFSAPI_MUTLIADDRESS):
         # ConnectionError/AttributeError if IPFS daemon not running
         client.id()
         return True
-    except (ConnectionError):  # exceptions.AttributeError):
-        logError("Daemon is not running at " + daemonMultiaddr)
+
+    except ConnectionError:  # exceptions.AttributeError):
+        logger.error('Daemon is not running at %s', daemonMultiaddr)
         return False
-    except OSError:
-        logError("IPFS is likely not installed. "
-                 "See https://ipfs.io/docs/install/")
-        sys.exit()
-    except Exception as e:
-        logError('Unknown error in retrieving daemon status')
-        logError(sys.exc_info()[0])
 
+    except OSError:
+        raise Exception(
```

Comment:
The prior implementation exited; an unhandled Exception causes the program to do the same. So, no fundamental difference. But, using `raise ... from ...` prints the original stacktrace without any extra actions - just using the language's built-in mechanisms. That's why I prefer it to explicit `sys.exit()`.

What is the reason of re-raising the exception though? The same reason which probably motivated to catch these exceptions, in the first place - to provide a bit more friendly error message to the user, I believe.

The more proper way would be to create custom exception classes and to catch them at an upper level of the application converting to nice readable messages, but I believe this can be implemented a bit later.